### PR TITLE
Reorganize modules to match ghaf mainline stucture

### PR DESCRIPTION
**Pull request by buianhhuy96**
- Move and rename some modules

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -43,35 +49,37 @@
...
+        (import ./hardware/fmo-os-rugged-laptop-7330.nix)
+        (import ./hardware/fmo-os-rugged-laptop-7330-public.nix)
+        (import ./hardware/fmo-os-rugged-tablet-7230.nix)
+        (import ./hardware/fmo-os-rugged-tablet-7230-public.nix)
+      ] ++ map generateInstConfig [
+        (import ./installers/fmo-os-installer.nix)
+        (import ./installers/fmo-os-installer-public.nix)
+      ];
+
+      #TODO Fix this
```
Comment (jsvapiav) [inline flake.nix:78]: Leftovers?

File: modules/virtualization/microvm/vm.nix [hunk 2]
```diff
@@ -92,9 +93,10 @@
         microvm.qemu.bios.enable = false;
         microvm.storeDiskType = "squashfs";
 
-        imports = (import "${ghafOS}/modules/module-list.nix") ++ (import ../../fmo-module-list.nix);
+        imports = (import "${ghafOS}/modules/module-list.nix");
```
Comment (jsvapiav) [inline modules/virtualization/microvm/vm.nix:96]: How that works now?

File: flake.nix [hunk 3]
```diff
@@ -16,12 +16,18 @@
 
   inputs = rec {
     ghafOS.url = "github:tiiuae/ghaf";
+    
+    #
+    # Flake and repo structuring configurations
+    #
+    # Allows us to structure the flake with the NixOS module system
+    flake-parts = {
+      url = "github:hercules-ci/flake-parts";
```
Comment (jsvapiav) [inline flake.nix:25]: What this for?

Comment (buianhhuy96) [reply flake.nix:25]: This is to sync with ghaf mainline structure, where they modularize everything with `self.nixosModules`

File: modules/virtualization/microvm/vm.nix (hunk 2 above)
Comment (buianhhuy96) [reply modules/virtualization/microvm/vm.nix:96]: This works thanks to `self.nixosModules.fmo-configs` in line 44, which is define in the new `fmo-module-list.nix`

File: flake.nix (hunk 1 above)
Comment (buianhhuy96) [reply flake.nix:78]: Ok. I should fix this, this comment belongs to ghaf mainline, I think they may run into some issue

Comment (jsvapiav) [reply flake.nix]: Than please give more meaningful comment here like and put down here that it belongs to ghaf mainline

File: modules/virtualization/microvm/vm.nix (hunk 2 above)
Comment (jsvapiav) [reply modules/virtualization/microvm/vm.nix:96]: Oh, okay, thanks!

File: flake.nix (hunk 1 above)
Comment (buianhhuy96) [reply flake.nix]: After checking it, I think we can delete it and update it when the mainline changes. It is just a syntax where they want to use `import` so nothing really affects us

**Pull request closed without merging**